Description: Replace the global-menu-bar container with an invisible box that's
  packed into the layout.
Author: William Hua <attente@canonical.com>

Index: dev.trusty/src/chrome/browser/ui/gtk/browser_window_gtk.cc
===================================================================
--- dev.trusty.orig/src/chrome/browser/ui/gtk/browser_window_gtk.cc	2014-03-13 15:43:38.180219690 -0400
+++ dev.trusty/src/chrome/browser/ui/gtk/browser_window_gtk.cc	2014-03-13 15:43:38.072215863 -0400
@@ -1626,7 +1626,9 @@
   // GtkMenuBar it sees into the global menu bar. (It doesn't seem to check the
   // visibility of the GtkMenuBar, so we can just permanently hide it.)
   global_menu_bar_.reset(new GlobalMenuBar(browser_.get()));
-  gtk_container_add(GTK_CONTAINER(window_vbox_), global_menu_bar_->widget());
+  gtk_box_pack_start(GTK_BOX(window_vbox_), global_menu_bar_->widget(), FALSE,
+                     FALSE, 0);
+  gtk_widget_show(global_menu_bar_->widget());
 
   // The window container draws the custom browser frame.
   window_container_ = gtk_alignment_new(0.0, 0.0, 1.0, 1.0);
Index: dev.trusty/src/chrome/browser/ui/gtk/global_menu_bar.cc
===================================================================
--- dev.trusty.orig/src/chrome/browser/ui/gtk/global_menu_bar.cc	2014-03-13 15:43:38.180219690 -0400
+++ dev.trusty/src/chrome/browser/ui/gtk/global_menu_bar.cc	2014-03-13 15:43:38.116217422 -0400
@@ -158,7 +158,7 @@
   // instead remain in our widget hierarchy simply to be noticed by third party
   // components.
   g_object_ref_sink(menu_bar_);
-  gtk_widget_set_no_show_all(menu_bar_, TRUE);
+  gtk_widget_set_size_request(menu_bar_, 0, 0);
 
   // Set a nice name so it shows up in gtkparasite and others.
   gtk_widget_set_name(menu_bar_, "chrome-hidden-global-menubar");
